contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,781 @@
+or `requiredDuringSchedulingIgnoredDuringExecution`).
+
+If `allowedAffinities` is totally absent from the spec, no affinity is allowed whatever its kind.
+
+#### Examples
+
+##### Basic policy
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  required:
+    affinities:
+      nodeAffinities:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - keys: ["beta.kubernetes.io/arch"]
+              operators: ["In"]
+              values: ["amd64", "arm64"]
+  allowed:
+    affinities:
+      nodeAffinities:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - keys: ["failure-domain.beta.kubernetes.io/region","kubernetes.io/authorized-region"]
+              operators: ["In","NotIn"]
+              values: ["eu-2", "us-1"]
+      podAntiAffinities: {}
+  default:
+    affinity:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - keys: ["beta.kubernetes.io/arch"]
+              operators: ["In"]
+              values: ["amd64"]
+```
+
+In this example, we allow:
+- hard NodeAffinity based on
+  -  `beta.kubernetes.io/arch` if value is `amd64` or `arm64`. Defaults to `amd64` if unspecified.
+  -  (optionally) any combination of <keys, operators, values> specified in `allowed` secion.
+- All podAntiAffinities
+- No podAffinities
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    affinities:
+      nodeAffinities:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - keys: ["failure-domain.beta.kubernetes.io/region","kubernetes.io/authorized-region"]
+              operators: ["In","NotIn"]
+              values: ["eu-2", "us-1"]
+          - matchExpressions:
+            - keys: ["failure-domain.beta.kubernetes.io/zone"]
+              operators: ["NotIn"]
+              values: ["dc1", "dc2"]
+          podAntiAffinities: {}
+```
+
+This example highlights the case where you don't want a full combinatory, here we allow the same affinities as the previous example, in addition,
+we allow a match expression that checks that some zone labels are not in the specified values.
+
+
+##### Allow-all policy
+In this example, all affinities are allowed:
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    affinities: {}
+```
+
+Which is equivalent to:
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    affinities:
+      nodeAffinities: {}
+      podAffinities: {}
+      podAntiAffinities: {}
+```
+
+If a sub-item of allowedAffinities is absent from SchedulingPolicy, it is not allowed e.g:
```

Comment:
This is confusing. You're saying that if no sub-items are specified they're all allowed, but as soon as you specify one, the others are implicitly denied?